engine/src/main/java/org/terasology/engine/world/generation/UpdatePriority.java

```diff
@@ -0,0 +1,43 @@
+// Copyright 2021 The Terasology Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+package org.terasology.engine.world.generation;
+
+/**
+ * Some preset priority values for Updates.priority.
+ */
+public final class UpdatePriority {
+    public static final int PRIORITY_PRODUCES = 250;
+    public static final int PRIORITY_CRITICAL = 200;
+    public static final int PRIORITY_HIGH = 150;
+    public static final int PRIORITY_NORMAL = 100;
+    public static final int PRIORITY_LOW = 50;
+    public static final int PRIORITY_TRIVIAL = 0;
+    public static final int PRIORITY_REQUIRES = -50;
```

Comment:
why not just use an enum?

Reply:
I'm just doing the same thing as [`EventPriority`](https://github.com/MovingBlocks/Terasology/blob/develop/engine/src/main/java/org/terasology/engine/entitySystem/event/EventPriority.java), I'm not sure why that's done that way.

Reply:
I'm pretty sure we're not using an enum for event priority because the main goal here is to resolve non-determinism due to same priority. In many cases these default threshold already help to sort out whether something should run with low or high priority, but we may need the fine-grained support to run a `NORMAL + 10` (slightly earlier than the rest of normal prio, but after high prio). 

Given that, I'm wondering whether the defaults for PRODUCES and REQUIRES are good as they are here, or whether they should be way higher/lower (maybe even Integer.MIN` and `Integer.MAX`).

Reply:
> Given that, I'm wondering whether the defaults for PRODUCES and REQUIRES are good as they are here, or whether they should be way higher/lower (maybe even Integer.MINandInteger.MAX`).

I think that makes sense for `PRODUCES`, since it's impossible for anything to update the facet before it's produced. But some things actually want to run after `REQUIRES`, specifically the `SpawnPlateauProvider` from `CoreWorlds`, although those conflicts can often also be fixed by changing the `Requires` to `Updates` on other providers  (`SimplexRoughnessProvider` in this case would pretend to update `ElevationFacet`).

Reply:
That totally makes sense to me. Is it possible to annotate the priority in `@Requires` as well?

Reply:
No, the idea is that if you need fine-grained control over when something reads the facet, you should just use `@Updates`. Of course, that is somewhat inconsistent with being able to `@Update` after `@Require`-ing like I did with the `SpawnPlateauProvider`, but I'm not sure what the best solution would be - it feels like any way of doing this is bad for modularity.